shogun-lib/src/main/java/de/terrestris/shoguncore/service/UserService.java

```diff
@@ -150,6 +158,45 @@ private void enableUser(User user) {
         repository.save(user);
     }
 
+    /**
+     * Changes user Password
+     * @param userId the ID of the user that is requesting a password change
+     * @param passwordChangeBody the request body, containing the old and new password
+     * @return
+     */
+    public void changeUserPassword(Long userId, PasswordChange passwordChangeBody) {
+        Optional<User> user = repository.findById(userId);
+
+        if (user.isPresent()) {
+            String currentPassword = user.get().getPassword();
+            String givenOldPassword = passwordChangeBody.getOldPassword();
+            if (passwordEncoder.matches(givenOldPassword, currentPassword)) {
+                String newPassword = passwordEncoder.encode(passwordChangeBody.getNewPassword());
+                user.get().setPassword(newPassword);
+                repository.save(user.get());
+            } else {
+                LOG.debug("Your current password does not match with the given old one. Aborting password change.");
+                throw new ResponseStatusException(
+                    HttpStatus.BAD_REQUEST,
+                    messageSource.getMessage(
+                        "password.change.OLD_PASSWORD_DOES_NOT_MATCH_ERROR",
+                        null,
+                        LocaleContextHolder.getLocale()
+                    )
+                );
+            }
+        } else {
+            LOG.debug("User not found. Aborting password change.");
+            throw new ResponseStatusException(
+                HttpStatus.BAD_REQUEST,
+                messageSource.getMessage(
+                    "password.change.USER_NOT_FOUND_ERROR",
+                    null,
+                    LocaleContextHolder.getLocale()
+                )
+            );
+        }
+    }
```

Comment:
The error handling (while throwing HTTP status exceptions) should take place in the controller.